test/lexer/scanSingleToken.ts

```diff
@@ -84,19 +84,19 @@ describe('src/lexer/scan', () => {
       const token = scanSingleToken(state, Context.None, 0);
       t.deepEqual(token, Token.Add);
     });
-    it('should recognise single character numbers', () => {
+    it('should recognize single character numbers', () => {
       const src = '3';
       const state = create(src, '', undefined);
       const token = scanSingleToken(state, Context.None, 0);
-      t.deepEqual(token, Token.NumericLiteral);
-      t.deepEqual(state.tokenValue, src);
+      t.equal(token, Token.NumericLiteral);
+      t.equal(state.tokenValue, Number(src));
     });
-    it('should recognise multi character numbers', () => {
+    it('should recognize multi character numbers', () => {
       const src = '345';
       const state = create(src, '', undefined);
       const token = scanSingleToken(state, Context.None, 0);
-      t.deepEqual(token, Token.NumericLiteral);
-      t.deepEqual(state.tokenValue, src);
```

Comment:
We should update these `.deepEqual` to `.equal` in future, they are just primitive values.